### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by davidgraeff**
Update Readme with build instructions etc. Roughly a copy of ESH readme, but a good first start, I guess.

Signed-off-by: davidgraeff <david.graeff@web.de>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,44 @@
...
+
+3\. Building with Maven
+=======================
+
+To build ths project from the sources, Maven takes care of everything:
+- set MAVEN_OPTS to "-Xms512m -Xmx1024m"
+- change into the openhab-core directory ("cd openhab-core“)
+- run "mvn clean install" to compile and package all sources
+
+If there are tests that are failing occasionally on your local build, run `mvn -DskipTests=true clean install` instead to skip them.
```
Comment (maggu2810) [inline README.md]: If there are tests that are failing occasionally on your local build ...

... fix / harden them. :wink:

Comment (davidgraeff) [reply README.md]: Haha I thought the same and was going to remove that sentence. And then I thought.. what happens if tests indeed fail, maybe at least give the user an option to skip them, I guess.

**Comment by davidgraeff** [comment]
This text is also very command line biased. It should be mentioned in another PR that maven-capable IDEs like Eclipse and Idea can also open the root pom and build the project.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,44 @@
...
+============
+
+Checkout the source code from GitHub, e.g. by running
+
+git clone https://github.com/openhab/openhab-core.git
+
+3\. Building with Maven
+=======================
+
+To build ths project from the sources, Maven takes care of everything:
```
Comment (cweitkamp) [inline README.md]: Typo: "ths project"

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidgraeff) [reply README.md]: This is a command line instruction. Single back ticks are correct, I think.

**Pull request merged**